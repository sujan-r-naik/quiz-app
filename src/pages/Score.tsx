import { Box, Typography } from "@mui/material";
import { Chart as ChartJS, ArcElement, Tooltip, Legend } from "chart.js";
import { Pie } from "react-chartjs-2";
import { useLocation } from "react-router-dom";

ChartJS.register(ArcElement, Tooltip, Legend);

type LocationStateProps = {
  trues: number;
  ans: any[];
};

const Score = () => {
  const location = useLocation();
  const state = location?.state as LocationStateProps;

  return (
    <Box
      sx={{
        height: "100vh",
        width: "100vw",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Box
        sx={{
          height: "100%",
          width: "49vw",
          overflowY: "scroll",
          p: 1,
        }}
      >
        {state.ans.map((x: any, i: number) => (
          <Box
            key={i}
            sx={{
              width: "42vw",
              my: 3,
              backgroundColor:
                x.a[0].isCorrect === x.ca[0].isCorrect ? "#d6ffde" : "#ffd6d6",
              borderRadius: 1,
              border: "solid 1px #777",
              p: 1,
            }}
          >
            <Typography
              role={`Display-Question-${i}`}
              sx={{ fontSize: "1.4rem", fontWeight: 600 }}
            >
              {x.q}
            </Typography>
            <Typography role={`Display-Correct-Answer-${i}`}>
              Correct Answer:{" "}
              {i === 4 ? JSON.stringify(x.a[0].answerText) : x.a[0].answerText}
            </Typography>
            <Typography role={`Display-Chosen-Answer-${i}`}>
              Your Answer:
              {i === 4
                ? JSON.stringify(x.ca[0].answerText)
                : x.ca[0].answerText}
            </Typography>
          </Box>
        ))}
      </Box>
      <Box
        sx={{
          height: "100%",
          width: "49vw",
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          p: 7,
        }}
      >
        <Pie
          role="score-piechart"
          data={{
            labels: ["Correct", "Wrong"],
            datasets: [
              {
                label: "Quiz Answers",
                data: [state.trues, 5 - state.trues],
                backgroundColor: ["#d6ffde", "#ffd6d6"],
                borderColor: ["green", "red"],
                borderWidth: 1,
              },
            ],
          }}
        />
      </Box>
    </Box>
  );
};

export default Score;
