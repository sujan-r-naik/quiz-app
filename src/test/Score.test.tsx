import "@testing-library/jest-dom/extend-expect";
import { screen, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Score from "../pages/Score";
import renderer from "react-test-renderer";

let answer = {
  qi: 1,
  q: "console.log() is used to read input from the user",
  a: [
    {
      answerText: "False",
      isCorrect: true,
    },
  ],
  ca: [
    {
      answerText: "False",
      isCorrect: true,
    },
  ],
};

const mockr = {
  pathname: "/score",
  hash: "",
  search: "",
  key: "",
  state: {
    trues: 3,
    ans: [answer, answer],
  },
};

test("Snapshot of Score component", () => {
  const comp = renderer.create(
    <MemoryRouter initialEntries={[mockr]}>
      <Score />
    </MemoryRouter>
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("In Score,", () => {
  test("does it render", () => {
    render(
      <MemoryRouter initialEntries={[mockr]}>
        <Score />
      </MemoryRouter>
    );

    let btnNext = screen.getByRole("Display-Question-0");
    expect(btnNext).toBeInTheDocument();

    let btnPrev = screen.getByRole("Display-Correct-Answer-0");
    expect(btnPrev).toBeInTheDocument();
  });
});
