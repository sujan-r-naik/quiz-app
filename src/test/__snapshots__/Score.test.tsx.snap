// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Score component 1`] = `
<div
  className="MuiBox-root css-1gsqmhk"
>
  <div
    className="MuiBox-root css-1m8kvpl"
  >
    <div
      className="MuiBox-root css-118x3hb"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-90qbru-MuiTypography-root"
        role="Display-Question-0"
      >
        console.log() is used to read input from the user
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        role="Display-Correct-Answer-0"
      >
        Correct Answer:
         
        False
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        role="Display-Chosen-Answer-0"
      >
        Your Answer:
        False
      </p>
    </div>
    <div
      className="MuiBox-root css-118x3hb"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-90qbru-MuiTypography-root"
        role="Display-Question-1"
      >
        console.log() is used to read input from the user
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        role="Display-Correct-Answer-1"
      >
        Correct Answer:
         
        False
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        role="Display-Chosen-Answer-1"
      >
        Your Answer:
        False
      </p>
    </div>
  </div>
  <div
    className="MuiBox-root css-np0z5b"
  >
    <canvas
      height={150}
      role="score-piechart"
      width={300}
    />
  </div>
</div>
`;
